proc/breakpoints.go

```diff
@@ -139,6 +139,11 @@ func (dbp *Process) setBreakpoint(tid int, addr uint64, temp bool) (*Breakpoint,
 	return dbp.Breakpoints[addr], nil
 }
 
+func (dbp *Process) writeSoftwareBreakpoint(thread *Thread, addr uint64) error {
```

Comment:
I like this, but rename to setSoftwareBreakpoint to be more consistent with setHardwareBreakpoint.